File: server/models/Appointment.js
const mongoose = require('mongoose');

const AppointmentSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.ObjectId,
    ref: 'User'
  },
  carId: {
    type: mongoose.Schema.ObjectId,
    ref: 'Car'
  },
  appointmentType: {
    type: String,
    required: [true, 'Please specify appointment type'],
    enum: ['test-drive', 'service', 'consultation']
  },
  appointmentDate: {
    type: Date,
    required: [true, 'Please specify appointment date and time']
  },
  status: {
    type: String,
    enum: ['scheduled', 'confirmed', 'completed', 'cancelled'],
    default: 'scheduled'
  },
  notes: {
    type: String
  },
  contactDetails: {
    name: {
      type: String,
      required: [true, 'Please provide contact name']
    },
    email: {
      type: String,
      required: [true, 'Please provide contact email'],
      match: [
        /^([\w-\.]+@([\w-]+\.)+[\w-]{2,4})?$/,
        'Please provide a valid email'
      ]
    },
    phone: {
      type: String,
      required: [true, 'Please provide contact phone']
    }
  },
  createdAt: {
    type: Date,
    default: Date.now
  },
  updatedAt: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: true
});

module.exports = mongoose.model('Appointment', AppointmentSchema);

File: fastapi-backend/app/schemas/appointment.py
from typing import Optional
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime
from enum import Enum

class AppointmentType(str, Enum):
    test_drive = "test-drive"
    service = "service"
    consultation = "consultation"

class AppointmentStatus(str, Enum):
    scheduled = "scheduled"
    confirmed = "confirmed"
    completed = "completed"
    cancelled = "cancelled"

class ContactDetails(BaseModel):
    name: str
    email: EmailStr
    phone: str

class AppointmentBase(BaseModel):
    userId: Optional[str] = None
    carId: Optional[str] = None
    appointmentType: AppointmentType
    appointmentDate: datetime
    status: AppointmentStatus = AppointmentStatus.scheduled
    notes: Optional[str] = None
    contactDetails: ContactDetails

class AppointmentCreate(AppointmentBase):
    pass

class AppointmentUpdate(BaseModel):
    appointmentType: Optional[AppointmentType] = None
    appointmentDate: Optional[datetime] = None
    status: Optional[AppointmentStatus] = None
    notes: Optional[str] = None
    contactDetails: Optional[ContactDetails] = None

class AppointmentInDB(AppointmentBase):
    id: str
    createdAt: datetime
    updatedAt: datetime

    class Config:
        orm_mode = True

class Appointment(AppointmentInDB):
    pass